### PR TITLE
"Learn More" link jumps to the associated troubleshooting paragraph directly

### PR DESCRIPTION
**Pull request by testforstephen**
The requirement is from issue https://github.com/Microsoft/vscode-java-debug/issues/360

### CODE REVIEW

File: Troubleshooting.md [hunk 1]
```diff
@@ -18,6 +18,11 @@ The debugger works with [Language Support for Java(TM) by Red Hat](https://marke
 
 ## Failed to resolve classpath:
 ### Reason:
+Below are the common failure reasons.
```
Comment (yaohaizh) [inline Troubleshooting.md]: In the document, it should avoid using `xyz` which is not meaningful. 

Should use meaningful name such as `mainClassName`, `projectName`

Comment (testforstephen) [reply Troubleshooting.md]: Nice suggestion. thanks.

File: src/utility.ts [hunk 2]
```diff
@@ -43,26 +44,27 @@ export async function showErrorMessage(message: IMessage, ...items: string[]): P
...
 
 function openLink(url: string): void {
     opn(url);
 }
 
-function handleTroubleshooting(choice: string, message: string): string | undefined {
+function handleTroubleshooting(choice: string, message: string, anchor: string): string | undefined {
     if (choice === LEARN_MORE) {
-        openLink(TROUBLESHOOTING_LINK);
+        const link = anchor ? `${TROUBLESHOOTING_LINK}#${anchor}` : TROUBLESHOOTING_LINK;
```
Comment (andxu) [inline src/utility.ts]: no need for a `link` variable since it is simple and used only once

File: src/utility.ts [hunk 3]
```diff
@@ -43,26 +44,27 @@ export async function showErrorMessage(message: IMessage, ...items: string[]): P
...
 export async function showWarningMessageWithTroubleshooting(message: IMessage, ...items: string[]): Promise<string | undefined> {
     const choice = await showWarningMessage(message, ...items, LEARN_MORE);
-    return handleTroubleshooting(choice, message.message);
+    return handleTroubleshooting(choice, message.message, message.anchor);
 }
 
 export async function showErrorMessageWithTroubleshooting(message: IMessage, ...items: string[]): Promise<string | undefined> {
     const choice = await showErrorMessage(message, ...items, LEARN_MORE);
-    return handleTroubleshooting(choice, message.message);
+    return handleTroubleshooting(choice, message.message, message.anchor);
```
Comment (andxu) [inline src/utility.ts]: rename IMessage/message  to  ITroubleShooting/troubleShoot?

File: src/utility.ts (hunk 2 above)
Comment (testforstephen) [reply src/utility.ts]: fixed.

File: src/utility.ts (hunk 3 above)
Comment (testforstephen) [reply src/utility.ts]: fixed.

**Pull request merged**